src/node/transaction.cpp

```diff
@@ -55,15 +55,17 @@ TransactionError BroadcastTransaction(const CTransactionRef tx, uint256& hashTx,
         }
     } else if (fHaveChain) {
         return TransactionError::ALREADY_IN_CHAIN;
-    } else {
+    } else if (wait_callback) {
         // Make sure we don't block forever if re-sending
         // a transaction already in mempool.
         promise.set_value();
     }
 
     } // cs_main
 
-    promise.get_future().wait();
+    if (wait_callback) {
```

Comment:
I think you can make this slightly clearer by adding a local variable `callback_set{false}`, setting it to `true` in the first `else if (wait_callback)` block, removing the second `else if (wait_callback)` block, and then testing for it here.

Reply:
Okay, also add params comments on it in.h